server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java

```diff
@@ -97,6 +97,23 @@ public T get(Settings settings) {
         }
     }
 
+    /**
+     * Returns the digest of this secure setting's value. This method can be called at all times, there is no requirement that
+     * {@code SecureSettings} be open, unlike {@code #get(Settings)}. The digest is used to check for changes of the value, without
+     * actually storing the value.
+     */
+    public byte[] getSecretDigest(Settings settings) {
+        final SecureSettings secureSettings = settings.getSecureSettings();
+        if (secureSettings == null || false == secureSettings.getSettingNames().contains(getKey())) {
```

Comment:
Shouldn't this be an error?

Reply:
I don't think so...
My rationale is that if a secure setting is missing its digest is *null* (similar to how a missing secure setting's value, without a fallback, is the empty secure string).
I've made this clear in the javadoc. I think *null* is appropriate to signal a missing value (given the codebase) but if you think otherwise (eg empty byte[]), I'll follow that suggestion.

Reply:
Well, it prob could not happen in the way this method is called right now, but its interface accepts a `Settings` object, so it's perfectly valid for a keystore to not have a value for each `SecureSetting` that we have in the code. Returning `null` here is about respecting the interface, not necessarily relevant in the algorithm right now.